### PR TITLE
Update copyright headers

### PR DESCRIPTION
**Pull request by domna**
This updates the copyright headers to 2024 as generated by `nyaml2nxdl`

### CODE REVIEW

File: base_classes/NXroot.nxdl.xml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 <?xml-stylesheet type="text/xsl" href="nxdlformat.xsl"?>
 <!--
 # NeXus - Neutron and X-ray Common Data Format
-# 
-# Copyright (C) 2014-2023 NeXus International Advisory Committee (NIAC)
-# 
+#
+# Copyright (C) 2014-2024 NeXus International Advisory Committee (NIAC)
```
Comment (lukaspie) [inline base_classes/NXroot.nxdl.xml:6]: How about the starting dates? NXroot has a starting year of 2008 in the currently accepted NIAC [proposal](https://github.com/nexusformat/definitions/blob/main/base_classes/NXroot.nxdl.xml). I don't think we should change this.

See also [this comment](https://github.com/nexusformat/definitions/pull/1303#issuecomment-1807711833) in out original nyaml2nxdl PR on the NIAC repo.

**Review by lukaspie — CHANGES_REQUESTED** [review]
Thanks for the update. Two things:
1) Is there a good reason why only our recent changes to the contributed classes received the new year? IIRC, NIAC has a routine to set the new finishing date for *all* classes.
2) We should make sure that we don't change the starting date for any of the existing classes (e.g. NXroot).

File: base_classes/NXroot.nxdl.xml (hunk 1 above)
Comment (domna) [reply base_classes/NXroot.nxdl.xml:6]: There is a discussion inside the NIAC on how to handle these. This PR is not intended to change the starting date or act as a platform for discussing this. It's just meant for updating the files to what `nyaml2nxdl` currently produces. We should then follow up on the general discussion on how to handle and generate copyright dates at a later stage.

Comment (lukaspie) [reply base_classes/NXroot.nxdl.xml:6]: Ok, makes sense. But we should ensure in the future that `nyaml2nxdl` picks up the correct starting date in agreement with what NIAC is proposing.

What about the first point, i.e., why are only the contributed classes changed? From [this line](https://github.com/FAIRmat-NFDI/nyaml/blob/1437afafd5a10e55121050e764bcf3c883c53f46/nyaml/nyaml2nxdl.py#L53), I would have guesses that all classes are updated to a new datetime? Or is this related to the fact that the yaml files did not change?

**Comment by domna** [comment]
> Thanks for the update. Two things:
> 
> 1. Is there a good reason why only our recent changes to the contributed classes received the new year? IIRC, NIAC has a routine to set the new finishing date for _all_ classes.

Actually, I don't completely know. I guess this depends on whether `nyaml2nxdl` uses the commented version when it was generated from nxdl vs. is completely generated from scratch.
I think we should have a discussion sometime on how to handle this, because I think there are some strange design decisions in how `nyaml2nxdl` handles this.

> 2. We should make sure that we don't change the starting date for any of the existing classes (e.g. NXroot).

The starting dates are already changed in our repository for some files. I don't know the latest status of the copyright discussion. This PR just updates the current year as generated by `nyaml2nxdl` to fix ci/cd for the whole repo.

My suggestion: We merge this to fix nxdl <-> nyaml ci/cd and open an issue on the general topic for later discussion. Sounds good?

**Comment by lukaspie** [comment]
Sounds good, you can merge this PR. I have created a new issues (#139) to keep track of the changed starting dates.

**Review by lukaspie — APPROVED** [review]
LGTM

**Pull request merged**